src/lib/components/objects/GLTF.svelte

```diff
@@ -0,0 +1,65 @@
+<script>
+	import { onMount } from 'svelte';
+
+	import * as THREE from 'three';
+	import { GLTFLoader } from 'three/examples/jsm/loaders/GLTFLoader.js';
+	import { DRACOLoader } from 'three/examples/jsm/loaders/DRACOLoader.js';
+    import { KTX2Loader } from 'three/examples/jsm/loaders/KTX2Loader';
+	import { setup } from '../../utils/context.js';
+	import { transform } from '../../utils/object.js';
+	import * as defaults from '../../utils/defaults.js';
+
+	/** @type {string} */
+	export let path = null;
+	/** @type {string} */
+	export let filename = null;
+	export let useDraco = false;
+	export let useKTX2 = false;
+
+	export let position = defaults.position;
+	export let rotation = defaults.rotation;
+	export let scale = defaults.scale;
+	export let castShadow = false;
+	export let receiveShadow = false;
+	export let frustumCulled = true;
+	export let renderOrder = 0;
+
+	/** @type {Object} */
+	let gltfScene = null;
+
+	const { root, parent, self } = setup(new THREE.Object3D());
+
+	$: {
+		if(gltfScene) {
+			gltfScene.traverse( function( node ) {
+				if ( node.isMesh ) { 
+					node.castShadow = castShadow;
+					node.receiveShadow = receiveShadow;
+					node.frustumCulled = frustumCulled;
+					node.renderOrder = renderOrder;
+				}
+			} );
+		}
+
+		transform(self, position, rotation, scale);
+		root.invalidate();
+	}
+
+	onMount(() => {
+
+		const loader = new GLTFLoader()
```

Comment:
Is there a benefit to creating the loader for each component instance rather than reusing?

Reply:
I hear what your saying but also just doing a thing and getting it off the ground/running is step # 1

Reply:
And reviewing it and iterating on it is step #2!

Reply:
Absolutely!  I submitted this PR early to get feedback, so I'm excited to get feedback.  I looked into the loader and figured out how to share it; i'll push a version with this in a bit.  This also shares the KTX and Draco loaders.

Reply:
ah! all good then. looking forward to previewing the deploy when that comes together